genesyscloud/outbound_campaign/resource_genesyscloud_outbound_campaign.go

```diff
@@ -0,0 +1,226 @@
+package outbound_campaign
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/v115/platformclientv2"
+	"log"
+	gcloud "terraform-provider-genesyscloud/genesyscloud"
+	"terraform-provider-genesyscloud/genesyscloud/consistency_checker"
+	resourceExporter "terraform-provider-genesyscloud/genesyscloud/resource_exporter"
+	"terraform-provider-genesyscloud/genesyscloud/util/resourcedata"
+	"time"
+)
+
+/*
+The resource_genesyscloud_outbound_campaign.go contains all of the methods that perform the core logic for a resource.
+*/
+
+// getAllAuthOutboundCampaign retrieves all of the outbound campaign via Terraform in the Genesys Cloud and is used for the exporter
+func getAllAuthOutboundCampaign(ctx context.Context, clientConfig *platformclientv2.Configuration) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
+	resources := make(resourceExporter.ResourceIDMetaMap)
+	proxy := getOutboundCampaignProxy(clientConfig)
+
+	campaigns, err := proxy.getAllOutboundCampaign(ctx)
+	if err != nil {
+		return nil, diag.Errorf("Failed to get campaigns: %s", err)
+	}
+
+	for _, campaign := range *campaigns {
+		// If a campaign is "stopping" during the export process we may encounter an error when we read the campaign later, and it will stop the export.
+		// We will give the campaign time to stop here and skip any that won't stop in time
+		if *campaign.CampaignStatus == "stopping" {
+			fmt.Println("Campaign is stopping")
+			// Retry to give the campaign time to turn off
+			err := gcloud.WithRetries(ctx, 180*time.Second, func() *retry.RetryError {
```

Comment:
Is 3 minutes really the length of time we can wait for the stopping status to change.  Ouch!

Reply:
Now that you mention it, it might be a bit harsh